handlers/aws/utils.py

```diff
@@ -122,10 +122,54 @@ def wrapper(lambda_event: dict[str, Any], lambda_context: context_.Context) -> s
     return wrapper
 
 
+def discover_integration_scope(lambda_event: dict[str, Any], at_record: int) -> str:
+    if (
+        "Records" not in lambda_event
+        or len(lambda_event["Records"]) < at_record
+        or "body" not in lambda_event["Records"][at_record]
+    ):
+        return "generic"
+
+    body: str = lambda_event["Records"][at_record]["body"]
+    s3_object_key: str = ""
+    try:
+        json_body: dict[str, Any] = json.loads(body)
+
+        if isinstance(json_body, dict) and "Records" in json_body and len(json_body["Records"]) > 0:
+            if "s3" in json_body["Records"][0]:
+                s3_object_key = json_body["Records"][0]["s3"]["object"]["key"]
+    except Exception:
+        pass
+
+    if s3_object_key == "":
+        shared_logger.info("s3 object key is empty, dataset set to `generic`")
+        return "generic"
+    else:
+        if (
+            "/CloudTrail/" in s3_object_key
+            or "/CloudTrail-Digest/" in s3_object_key
+            or "/CloudTrail-Insight/" in s3_object_key
+        ):
+            return "aws.cloudtrail"
+        elif "exportedlogs" in s3_object_key or "awslogs" in s3_object_key:
+            return "aws.cloudwatch_logs"
+        elif "/elasticloadbalancing/" in s3_object_key:
+            return "aws.elb_logs"
+        elif "/network-firewall/" in s3_object_key:
+            return "aws.firewall_logs"
+        elif "/vpcflowlogs/" in s3_object_key:
+            return "aws.vpcflow"
+        elif "/WAFLogs/" in s3_object_key:
+            return "aws.waf"
+        else:
```

Comment:
I wonder if we should add a log /warning message here to tell user here are the 6 log type we support and the rest goes into generic.

Reply:
we could have but not here in my opinion. rather in the elasticsearch sipper: I will do